app/domain/export/tabular/people/person_row.rb

```diff
@@ -31,11 +31,12 @@ def roles
     end
 
     def entry_roles
-      if restrict_to_roles.present?
-        entry
-          .roles
-          .where(type: restrict_to_roles)
-          .where(group_id: restrict_to_group_ids)
+      if role_restrictions.present?
+        roles = entry.roles
+
+        role_restrictions.flat_map do |group_id, role_types|
+          role_types.map { |type| roles.find_by(group_id: group_id, type: type) }.compact
```

Comment:
Müsstest du nach `flat_map` nicht noch `uniq` machen? Wenn ich zwei Filter habe die dieselbe Rolle in derselben Gruppe filtern. Bei Personenfiltern geht das nicht, aber bei Abos schon: Zum Beispiel wenn ich alle Leitenden mit dem Tag x und in einer separaten Subscription alle Leitenden und Mitglieder mit dem Tag y einschliesse.